src/OpenRiaServices.Tools/Test/MsBuildHelper.cs

```diff
@@ -150,8 +152,11 @@ internal static ProjectWrapper LoadProject(string projectPath)
                     }
                     else
                     {
-                        // fallback to net472
-                        project.SetGlobalProperty("TargetFramework", "net472");
+                        var frameworks = targetFrameworks.Split(';');
+                        var version = (TargetFrameworkAttribute)Assembly.GetExecutingAssembly().GetCustomAttributes(typeof(System.Runtime.Versioning.TargetFrameworkAttribute), false).SingleOrDefault();
+                        var isNet472 = version.FrameworkName == ".NETFramework,Version=v4.7.2";
```

Comment:
I think you will always get net4** when running msbuild (such as inside visual studio)
and net6+ when running from dotnet 6.

Since we want visual studio to work with (net6) we might want have separate server class lib projects targeting net6 and net framework.

If we have much problem with this approach we might consider dropping an exe in the server folder (or call dotnet) so that the code generation runs as a separate "net6" process